### PR TITLE
New: Venmo payments via PayPal are now accepted

### LINKED ISSUE #6386: GiveWP should integrate with Venmo Payments
It would be good if we had an add-on to integrate GiveWP and Venmo Payments.

https://givewp.canny.io/admin/board/feature-requests/p/givewp-should-integrate-with-venmo-payments

### PR DESCRIPTION
**Pull request by kjohnson**
Resolves #6386 

## Description

This PR enabled Venmo as a PayPal funding source.

## Visuals

image

image

## Testing Instructions

- Enable Venmo for PayPal
## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: src/PaymentGateways/PayPalCommerce/PayPalCommerce.php [hunk 1]
```diff
@@ -160,6 +160,16 @@ public function getOptions()
...
+            [
+                'name' => esc_html__('Accept Venmo', 'give'),
+                'id' => 'paypal_commerce_accept_venmo',
+                'type' => 'radio_inline',
+                'default' => 'disabled',
+                'options' => [
+                    'enabled' => esc_html__('Enabled', 'give'),
+                    'disabled' => esc_html__('Disabled', 'give'),
+                ],
+            ],
```
Comment (LMNTL) [inline src/PaymentGateways/PayPalCommerce/PayPalCommerce.php:172]: Clarifying question: can admins with a PayPal account from outside of the US accept payments through Venmo?

**Review by LMNTL — CHANGES_REQUESTED** [review]
This looks good so far! When I click on the 'Venmo' button, I'm getting the following error:
image

Here are the full contents of the error object;
```
{
  buttonCorrelationID: "f445453c4d7b2"
  buttonSessionID: "uid_853aa3298a_mty6nda6mdc"
  clientID: "AbMGETAnJeEeTFVJE_cEim_bROArQQkK0RAvvlIpUBEJ2w7bFkhZeyY5YT7FAYFGTW-9gEOvQz0onurd"
  env: "production"
  err: "Error: Expected reject to be called with Error, got [object Null]\n    at e.n.reject (https://www.paypal.com/sdk/js?client-id=AbMGETAnJeEeTFVJE_cEim_bROArQQkK0RAvvlIpUBEJ2w7bFkhZeyY5YT7FAYFGTW-9gEOvQz0onurd&merchant-id=BT7P6Q6TNS4JG&components=hosted-fields,buttons&locale=en_US&disable-funding=credit&enable-funding=venmo&vault=false&intent=capture&currency=USD:2:67322)\n    at https://www.paypal.com/sdk/js?client-id=AbMGETAnJeEeTFVJE_cEim_bROArQQkK0RAvvlIpUBEJ2w7bFkhZeyY5YT7FAYFGTW-9gEOvQz0onurd&merchant-id=BT7P6Q6TNS4JG&components=hosted-fields,buttons&locale=en_US&disable-funding=credit&enable-funding=venmo&vault=false&intent=capture&currency=USD:2:67915\n\nError: Expected reject to be called with Error, got [object Null]\n    at k.error (https://www.paypal.com/sdk/js?client-id=AbMGETAnJeEeTFVJE_cEim_bROArQQkK0RAvvlIpUBEJ2w7bFkhZeyY5YT7FAYFGTW-9gEOvQz0onurd&merchant-id=BT7P6Q6TNS4JG&components=hosted-fields,buttons&locale=en_US&disable-funding=credit&enable-funding=venmo&vault=false&intent=capture&currency=USD:2:159313)\n    at Object.<anonymous> (https://www.paypal.com/sdk/js?client-id=AbMGETAnJeEeTFVJE_cEim_bROArQQkK0RAvvlIpUBEJ2w7bFkhZeyY5YT7FAYFGTW-9gEOvQz0onurd&merchant-id=BT7P6Q6TNS4JG&components=hosted-fields,buttons&locale=en_US&disable-funding=credit&enable-funding=venmo&vault=false&intent=capture&currency=USD:2:172343)\n    at JSON.parse (<anonymous>)\n    at o (https://www.paypal.com/sdk/js?client-id=AbMGETAnJeEeTFVJE_cEim_bROArQQkK0RAvvlIpUBEJ2w7bFkhZeyY5YT7FAYFGTW-9gEOvQz0onurd&merchant-id=BT7P6Q6TNS4JG&components=hosted-fields,buttons&locale=en_US&disable-funding=credit&enable-funding=venmo&vault=false&intent=capture&currency=USD:2:172206)\n    at se (https://www.paypal.com/sdk/js?client-id=AbMGETAnJeEeTFVJE_cEim_bROArQQkK0RAvvlIpUBEJ2w7bFkhZeyY5YT7FAYFGTW-9gEOvQz0onurd&merchant-id=BT7P6Q6TNS4JG&components=hosted-fields,buttons&locale=en_US&disable-funding=credit&enable-funding=venmo&vault=false&intent=capture&currency=USD:2:172355)\n    at u.on (https://www.paypal.com/sdk/js?client-id=AbMGETAnJeEeTFVJE_cEim_bROArQQkK0RAvvlIpUBEJ2w7bFkhZeyY5YT7FAYFGTW-9gEOvQz0onurd&merchant-id=BT7P6Q6TNS4JG&components=hosted-fields,buttons&locale=en_US&disable-funding=credit&enable-funding=venmo&vault=false&intent=capture&currency=USD:2:177015)\n    at ve (https://www.paypal.com/sdk/js?client-id=AbMGETAnJeEeTFVJE_cEim_bROArQQkK0RAvvlIpUBEJ2w7bFkhZeyY5YT7FAYFGTW-9gEOvQz0onurd&merchant-id=BT7P6Q6TNS4JG&components=hosted-fields,buttons&locale=en_US&disable-funding=credit&enable-funding=venmo&vault=false&intent=capture&currency=USD:2:177134)\n    at https://www.paypal.com/sdk/js?client-id=AbMGETAnJeEeTFVJE_cEim_bROArQQkK0RAvvlIpUBEJ2w7bFkhZeyY5YT7FAYFGTW-9gEOvQz0onurd&merchant-id=BT7P6Q6TNS4JG&components=hosted-fields,buttons&locale=en_US&disable-funding=credit&enable-funding=venmo&vault=false&intent=capture&currency=USD:2:183838\n    at Function.e.try (https://www.paypal.com/sdk/js?client-id=AbMGETAnJeEeTFVJE_cEim_bROArQQkK0RAvvlIpUBEJ2w7bFkhZeyY5YT7FAYFGTW-9gEOvQz0onurd&merchant-id=BT7P6Q6TNS4JG&components=hosted-fields,buttons&locale=en_US&disable-funding=credit&enable-funding=venmo&vault=false&intent=capture&currency=USD:2:70625)\n    at https://www.paypal.com/sdk/js?client-id=AbMGETAnJeEeTFVJE_cEim_bROArQQkK0RAvvlIpUBEJ2w7bFkhZeyY5YT7FAYFGTW-9gEOvQz0onurd&merchant-id=BT7P6Q6TNS4JG&components=hosted-fields,buttons&locale=en_US&disable-funding=credit&enable-funding=venmo&vault=false&intent=capture&currency=USD:2:183588"
  referer: "www.paypal.com"
  sdkCorrelationID: "f389501e9f269"
  sessionID: "uid_e49b3db7d5_mty6mze6mzg"
  time: "1651077616957"
  timestamp: "1651077616957"
  token: null
  user_id: "uid_853aa3298a_mty6nda6mdc"
}
```

Happening on Chrome 100.0.4896.127. I get variants of the same message in both test and live mode.

**Comment by JasonTheAdams** [comment]
@kjohnson @LMNTL This is pretty high priority. Can we please prioritize getting this working?

**Comment by kjohnson** [comment]
@LMNTL hm.. I made no additional configuration

image

**Comment by kjohnson** [comment]
Interestingly, I see it working in FireFox, but not Brave.

**Comment by kjohnson** [comment]
@LMNTL looks like the error you are seeing is a separate bug with PayPal Commerce from validation. I'm passing that along to @ravinderk for consideration. It seems that our PayPal code isn't handling the email validation correctly.

**Review by LMNTL — APPROVED** [review]

**Comment by LMNTL** [comment]
@kjohnson @JasonTheAdams Ah, you are absolutely right, I should've checked it against `develop` first. Approving this since it was an unrelated issue. Strangely, I'm able to reproduce it even when filling a real email address in the form.

**Pull request merged**